api/friendsLists.js

```diff
@@ -1 +1,98 @@
-//friendsLists file - delete this if you're writing this file
\ No newline at end of file
+const express = require("express");
+const friendsRouter = express.Router();
+
+const {
+  addFriends,
+  getUserByUsername,
+  removeFriend,
+  getFriendsByUserId,
+  getPostsByUserId,
+} = require("../db");
+
+const { requireUser } = require("./utils");
+
+friendsRouter.post(
+  "/friends/:friendId",
+  requireUser,
+  async (req, res, next) => {
+    try {
+      const { username } = req.params;
+      const { id: userId } = req.user;
+      const { id: friendsId } = await getUserByUsername(username);
```

Comment:
Destructuring from a request that can fail can cause the server to crash if this fails - so if a user provides a bad username it would crash the server. When getting a result you first need to make sure that the request succeeds before accessing it's properties. 
- All that being said but we can simply just change the endpoint to have the username in it and then we don't need to fetch it.